### PR TITLE
Add unit tests for freeze transactions

### LINKED ISSUE #302: Unit test for system transactions logging
**Detailed Description**
There are no unit tests for contract transactions logging

**Actual Behavior**
Run unit test, no tests are run

**Expected Behavior**
Tests for system transactions are run
* Freeze
* System Delete/undelete

**Environment:**
Version: 0.2.0.rc4

**Additional Context**
Add any other context about the problem here. Attach any logs here, if applicable.

### PR DESCRIPTION
**Pull request by gregscullard**
Signed-off-by: Greg Scullard <gregscullard@hedera.com>

**Detailed description**:
Adds unit tests for freeze transactions

**Which issue(s) this PR fixes**:
Fixes #302 

**Checklist**
- [ ] Documentation added
- [X] Tests updated

### CODE REVIEW

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
+package com.hedera.recordLogger;
+
+import com.google.protobuf.ByteString;
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java]: Move imports below license header

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
+package com.hedera.recordLogger;
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java]: Wrong package

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+import java.time.Instant;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Sql("classpath:db/scripts/cleanup.sql") // Class manually commits so have to manually cleanup tables
+public class RecordFileLoggerFreezeTest extends AbstractRecordFileLoggerTest {
+
+	//TODO: The following are not yet saved to the mirror node database
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java]: Remove

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+import org.junit.jupiter.api.*;
+import org.springframework.test.context.jdbc.Sql;
+import org.junit.jupiter.api.Test;
+import java.time.Instant;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Sql("classpath:db/scripts/cleanup.sql") // Class manually commits so have to manually cleanup tables
```
Comment (steven-sheehy) [inline src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java]: AFTER_TEST_METHOD?

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java (hunk 1 above)
Comment (gregscullard) [reply src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java]: Done

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java (hunk 2 above)
Comment (gregscullard) [reply src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java]: Done

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java (hunk 3 above)
Comment (gregscullard) [reply src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java]: Done

File: src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java (hunk 4 above)
Comment (gregscullard) [reply src/test/java/com/hedera/recordLogger/RecordFileLoggerFreezeTest.java]: Fixed to match other tests

**Comment by gregscullard** [comment]
Note, this PR depends on other PRs being merged (316). Failing CI tests are due to wider changes resulting from master merge which also affect other test PRs. Will review once 316 is approved and merged into this one.

**Comment by apeksharma** [comment]
Looks like #316 has been merged. Let's get these good set of tests in.

**Comment by mike-burrage-hedera** [comment]
Decide whether this needs to be reassigned for implementation or closed by COB MOnday.

**Comment by steven-sheehy** [comment]
This is ready for review. I also hopefully fixed another build error that I saw occur on other PRs as well with surefire maven plugin being unable to fork. Fix was to increase surefire memory and to not call `System.exit(0)` as it is incompatible with surefire. Here was the error:

```console
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project mirror-node: There are test failures.
[ERROR] 
[ERROR] Please refer to /root/project/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /root/project && /opt/java/openjdk/bin/java -javaagent:/root/.m2/repository/org/jacoco/org.jacoco.agent/0.8.4/org.jacoco.agent-0.8.4-runtime.jar=destfile=/root/project/target/jacoco.exec -jar /root/project/target/surefire/surefirebooter11354676293905463448.jar /root/project/target/surefire 2019-11-08T22-48-33_756-jvmRun1 surefire15138054201692368137tmp surefire_011393843503498543407tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 137
[ERROR] Crashed tests:
[ERROR] com.hedera.mirror.downloader.record.RecordFileDownloaderTest
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /root/project && /opt/java/openjdk/bin/java -javaagent:/root/.m2/repository/org/jacoco/org.jacoco.agent/0.8.4/org.jacoco.agent-0.8.4-runtime.jar=destfile=/root/project/target/jacoco.exec -jar /root/project/target/surefire/surefirebooter11354676293905463448.jar /root/project/target/surefire 2019-11-08T22-48-33_756-jvmRun1 surefire15138054201692368137tmp surefire_011393843503498543407tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 137
[ERROR] Crashed tests:
[ERROR] com.hedera.mirror.downloader.record.RecordFileDownloaderTest
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:669)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:282)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:245)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1183)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1011)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:857)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
[ERROR] 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
[ERROR] 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:61)
[ERROR] 
```

File: src/test/java/com/hedera/recordFileLogger/RecordFileLoggerFreezeTest.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    	final Transaction transaction = freezeTransaction();
+    	final TransactionBody transactionBody = TransactionBody.parseFrom(transaction.getBodyBytes());
+    	final TransactionRecord record = transactionRecord(transactionBody);
+
+    	RecordFileLogger.storeRecord(transaction, record);
+    	RecordFileLogger.completeFile("", "");
+
+    	final com.hedera.mirror.domain.Transaction dbTransaction = transactionRepository.findById(Utility.timeStampInNanos(record.getConsensusTimestamp())).get();
+
+    	assertAll(
```
Comment (apeksharma) [inline src/test/java/com/hedera/recordFileLogger/RecordFileLoggerFreezeTest.java]: looks good overall

There is so much copy-paste with minor changes of this assertAll block in the tests files derived from AbstractRecordFileLoggerTest, OMG!
I understand it's not by you, and therefore making other places better should not be just your responsibility, we can do it as team later.
For this file, can we move this 20 lines in separate function called, say `verifySuccessfullyRecorded()`, and use it in tests.

**Pull request merged**